docs/source/reference/executors.rst

```diff
@@ -0,0 +1,41 @@
+.. _`executor api`:
+
+LiberTEM executors
+==================
+
+See :ref:`executors` for a general overview.
+
+.. versionadded:: 0.9.0
+    Previously, the executor API was internal. Since influence on the executor
+    is important for integration with Dask and other frameworks,
+    the API is now documented to help with that. Nevertheless, this is still
+    experimental and may change between releases without notice.
```

Comment:
Can we move this documentation to the development section, and clearly mark it as an internal API? I think it would be a mistake to make the `JobExecutor` API public (or even quasi-public), especially in its current state. It may be important for integration with dask etc. - but what is important for the _user_ is not the internal executor API.

"still experimental" means to me that it is already usable but subject to change, and on the track to be stabilized. If I read that in the API docs of a library, I might start writing code (outside of that library!) against that API, and expect that to keep the same in principle, just with some minor changes. I don't think this can be expected from the `JobExecutor` API currently.

Making changes like #1143 in an efficient manner very much depends on keeping the API private, so we can change all implementations and users in lock-step. Generally, if we make an API available to users as experimental which is not on track to stabilize, it will create problems both on our side, and on the user side. We can't really make sweeping changes, because once there is user code out there, we rightfully have some commitment to keep it working or to make a reasonable migration possible, and on the user side, it will be very inconvenient if every new release means changing code, even if a migration is possible.

The user-facing parts are another story: the executors themselves should be documented, and which should be chosen in which situation, how they can be constructed (`DaskJobExecutor.{connect,make_local}` etc.).

If we want to think about making the executor API public (even experimental), we should wait until we are progressed further with #473, in my opinion - I don't think much of the current API will survive in its current form.

I hope this does not read as too argumentative - and to take a step back: 

- What is the user case for calling any of the methods on the `JobExecutor` interface, except for the constructor(s)?
- What is the use case for subclassing either `JobExectuor` or one of out implementations?
- Generally, what is the actual use case for making this API public?

TL;DR: #427

Reply:
Good points! I've pushed some changes. Does that sound better? Argumentative is good to get the best result together. :-)